components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/internal/OrganizationManagementDataHolder.java

```diff
@@ -96,10 +79,20 @@ public DataSource getDataSource() {
     public void initDataSource() {
 
         try {
-            this.dataSource = DatabaseUtil.getRealmDataSource(CarbonContext.getThreadLocalCarbonContext().
-                    getUserRealm().getRealmConfiguration());
```

Comment:
The initDataSource method is executed at the bundle activation phase and `CarbonContext.getThreadLocalCarbonContext().getUserRealm()` is null even at this stage. Assume the changes done for the service component by this PR leads to this issue and could not identity the root cause. Hence the real configurations were resolved as below by loading the super tenant's realm configurations.
As the organization management feature related data resides in SharedDB, and the super tenant realm configuration includes the relevant information, this workaround is working as expected.